packages/suite/src/utils/wallet/coinjoinUtils.ts

```diff
@@ -61,6 +61,45 @@ export const breakdownCoinjoinBalance = ({
     return balanceBreakdown;
 };
 
+/**
+ * Calculates account anonymity progress – how much UTXOs are anonymized relatively to the target anonymity
+ */
+export const calculateAnonymityProgress = ({
+    targetAnonymity,
+    anonymitySet,
+    utxos,
+}: {
+    targetAnonymity: number | undefined;
+    anonymitySet: AnonymitySet | undefined;
+    utxos: Account['utxo'];
+}): number => {
+    if (!anonymitySet || targetAnonymity === undefined || !utxos) {
+        return 0;
+    }
+
+    let weightedCurrentAnonymity = new BigNumber(0);
+    let weightedTargetAnonymity = new BigNumber(0);
+
+    utxos?.forEach(({ address, amount }) => {
+        const bigAmount = new BigNumber(amount);
+
+        const currentAnonymity = anonymitySet[address] || 0;
+
+        weightedCurrentAnonymity = weightedCurrentAnonymity.plus(
+            bigAmount.times(Math.min(currentAnonymity, targetAnonymity) - 1),
+        );
+        weightedTargetAnonymity = weightedTargetAnonymity.plus(
+            bigAmount.times(targetAnonymity - 1),
```

Comment:
I know it should not be reachable in our current flow, but still would be nice to handle the case when `targetAnonymity === 1`

```javascript
if(targetAnonymity === 1) {
  return 100;
}
```

Reply:
f56eaa083acc69a8652848b738da50547c3beaa6